python/ray/experimental/channel/common.py

```diff
@@ -29,7 +33,28 @@ def register_custom_serializer(self) -> None:
         default device. Instead, these should be extracted from the
         worker-local _SerializationContext.
         """
-        pass
+        if self._contains_type is not None:
+            self._contains_type.register_custom_serializer()
+
+    @property
+    def is_direct_return(self) -> bool:
+        """
+        Some channels may contain other values that should be sent via a
+        different channel. This returns whether the value is a direct return or
+        if it is "nested" inside a different channel.
+        """
+        return True
+
+    @property
+    def contains_type(self) -> "ChannelOutputType":
+        return self._contains_type
+
+    def set_contains_type(self, typ: "ChannelOutputType") -> None:
+        from ray.experimental.channel.torch_tensor_type import TorchTensorType
+
+        if typ is not None and not isinstance(typ, TorchTensorType):
+            raise ValueError("Contained type must be of type TorchTensorType")
```

Comment:
The only other type we have is SharedMemoryType, and it doesn't make sense for a SharedMemoryType to contain a SharedMemoryType. I'll change this to an assert since it should not be user-facing.